pkg/controller/controllerutil/controllerutil.go

```diff
@@ -32,6 +35,91 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 )
 
+var (
+	// Will store memoized controller namespace
+	inClusterNamespace      string
+	inClusterNamespacePath  = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
```

Comment:
could this be an alternative instead of declaring the path as a variable?
```go
var readServiceAccountFile = func() ([]byte, error) {
	return ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
}



func ReadInClusterNamespace() (string, error) {
      nsBytes, err :=  readServiceAccountFile()
      ......
}
```

this would make testing easier, in the sense:
```go
It("should return the right namespace", func() {
     // overwriting the variable
     readServiceAccountFile = func() ([]byte, error) {
          return []byte("testnamespace"), nil
     }

     ns, err := ReadInClusterNamespace()
     Expect(err).Should(BeNil())
     Expect(ns).To(Equal("testnamespace"))
}) 
```